src/server.rs

```diff
@@ -20,57 +20,112 @@ use tokio_service::NewService;
 /// A message from server to client.
 pub type Response<T, E> = Result<T, WireError<E>>;
 
-/// Spawns a service that binds to the given address and runs on the default reactor core.
-pub fn listen<S, Req, Resp, E>(addr: SocketAddr, new_service: S) -> ListenFuture
-    where S: NewService<Request = Result<Req, DeserializeError>,
-                        Response = Response<Resp, E>,
-                        Error = io::Error> + Send + 'static,
-          Req: Deserialize + 'static,
-          Resp: Serialize + 'static,
-          E: Serialize + 'static
-{
-    let (tx, rx) = futures::oneshot();
-    REMOTE.spawn(move |handle| Ok(tx.complete(listen_with(addr, new_service, handle.clone()))));
-    ListenFuture { inner: rx }
-}
+cfg_if! {
```

Comment:
if the `tls_acceptor` arg is replaced with a config struct, then we can probably constrain the `cfg_if!` to a much smaller surface of the code. It's fine if the config struct is empty (for now) when tls isn't enabled.